### PR TITLE
announce HHVM 4.68

### PR DESCRIPTION
**Pull request by jjergus**

### CODE REVIEW

File: _posts/2020-07-30-hhvm-4.68.markdown [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  to throw an exception.
+  - A `default` branch should be added to any `switch` statements that
+    intentionally don't cover all possible cases.
+  - Use the INI option `hhvm.throw_on_non_exhaustive_switch=0` to temporarily
+    disable this warning, or `hhvm.throw_on_non_exhaustive_switch=2` to enable
+    the future behavior now (throw an exception).
+- By default, HHVM now triggers an unrecoverable error if any function return
+  type annotations are violated.
+  - The INI option
+    [`hhvm.check_return_type_hints`](https://github.com/facebook/hhvm/blob/HHVM-4.68/hphp/runtime/base/runtime-option.h#L734)
```
Comment (lexidor) [inline _posts/2020-07-30-hhvm-4.68.markdown:83]: I tried on the latest nightly and `function _(): int { return ''; }` throws a `TypeError` when invoked.
Is the intent to behave like `\trigger_error('Typeviolation...', \E_ERROR);` which is truly unrecoverable to throw a TypeError, which only is unrecoverable when uncaught.

Comment (lexidor) [reply _posts/2020-07-30-hhvm-4.68.markdown:83]: https://github.com/facebook/hhvm/blob/master/hphp/runtime/base/runtime-error.cpp#L106-L114
```CPP

void raise_return_typehint_error(const std::string& msg) {
  if (RuntimeOption::PHP7_EngineExceptions) {
    VMRegAnchor _;
    SystemLib::throwTypeErrorObject(msg);
  }
  raise_recoverable_error(msg);
  raise_error("Error handler tried to recover from a return typehint violation");
}
```

Does someone know the default value of `RuntimeOption::PHP7_EngineExceptions`?

**Comment by fredemmott** [comment]
The option is on by default, but isn’t really as significant as it sounds: if it’s off, you can throw from the error handler instead, and catch at an outer scope.

What you can’t do is return true from your errror handler and continue in the same scope - like basic “on error resume next”.

File: _posts/2020-07-30-hhvm-4.68.markdown (hunk 1 above)
Comment (lexidor) [reply _posts/2020-07-30-hhvm-4.68.markdown:83]: > The option is on by default, but isn’t really as significant as it sounds: if it’s off, you can throw from the error handler instead, and catch at an outer scope.

Then the default does not match the blog post. The default only changes if you changed the engine exception runtime setting. The default remains a TypeError. I interpreted the blog post text as `hhvm will not throw a TypeError anymore and hard abort your request with a hard unrecoverable error.`

Comment (jjergus) [reply _posts/2020-07-30-hhvm-4.68.markdown:83]: Good catch, I'll clarify that.

**Pull request merged**